tests/ui/recursion/recursive-enum-box.rs

```diff
@@ -0,0 +1,12 @@
+//@ run-pass
+// A smoke test for recursive enum structures using Box<T>.
+// This test constructs a linked list-like structure to verify memory allocation and ownership.
```

Comment:
Nit: (we're not verifying here)

```suggestion
// This test constructs a linked list-like structure to exercise memory allocation and ownership.
```